runtime/admin/src/main/java/org/apache/polaris/admintool/config/QuarkusProducers.java

```diff
@@ -43,6 +44,12 @@ public MetaStoreManagerFactory metaStoreManagerFactory(
     return metaStoreManagerFactories.select(Identifier.Literal.of(persistenceType)).get();
   }
 
+  @Produces
+  @ApplicationScoped
+  public SchemaInitializer schemaInitializer(MetaStoreManagerFactory metaStoreManagerFactory) {
+    return (SchemaInitializer) metaStoreManagerFactory;
```

Comment:
Please use `.select()` as in `metaStoreManagerFactory()` for proper type handling.